#!/usr/bin/env python
PACKAGE="vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("low_h", int_t, 0, "Lower Bound Of H", 0, 0, 255);
gen.add("high_h", int_t, 0, "Higher Bound Of H", 255, 0, 255);

gen.add("low_s", int_t, 0, "Lower Bound Of S", 99, 0, 255);
gen.add("high_s", int_t, 0, "Higher Bound Of S", 205, 0, 255);

gen.add("low_v", int_t, 0, "Lower Bound Of V", 0, 0, 255);
gen.add("high_v", int_t, 0, "Higher Bound Of V", 172, 0, 255);

gen.add("opening_closing_mat_point", int_t, 0, "", 1, 1, 7);
gen.add("opening_iter", int_t, 0, "Opening iterations", 1, 1, 10);
gen.add("closing_iter", int_t, 0, "Closing iterations", 1, 1, 10);

exit(gen.generate(PACKAGE, "vision_tasks", "gateRange"));
